src/server/state.rs

```diff
@@ -109,8 +109,19 @@ impl State {
 
     pub fn process_task_failed(&mut self, msg: TaskFailedMessage) {
         log::debug!("Task id={} failed", msg.id);
+
         let job = self.get_job_mut_by_tako_task_id(msg.id).unwrap();
-        job.set_failed_state(msg.id, msg.info.message);
+        if job.counters.n_failed_tasks >= job.n_to_cancel_job {
+            let test = job.make_job_detail(true);
+            for x in test.tasks.iter() {
+                let (_, state) = job.get_task_state_mut(x.task_id.into());
+                if matches!(state ,JobTaskState::Running) || matches!(state, JobTaskState::Waiting) {
+                    job.set_cancel_state(x.task_id.into());
+                }
+            }
+        } else {
+            job.set_failed_state(msg.id, msg.info.message);
```

Comment:
* ``job.set_failed_state(msg.id, msg.info.message);`` should be always called even max_fail was reached